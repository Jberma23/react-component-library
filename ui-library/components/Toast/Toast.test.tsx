import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Toast } from ".";

describe(Toast, () => {
  it("renders with required props", () => {
    const { asFragment } = render(
      <Toast variant="success" dismissHandler={() => {}}>
        Hello World
      </Toast>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("has a clickable dismiss button", async () => {
    const { asFragment } = render(
      <Toast variant="success" dismissHandler={() => {}}>
        Hello World
      </Toast>
    );
    await userEvent.click(screen.getByTitle("dismiss notification"));
    expect(asFragment()).toMatchSnapshot();
  });

  // it("passes all a11y checks", async () => {
  //   await expect(
  //     <Toast variant="danger" dismissHandler={() => {}}>
  //       Hello World
  //     </Toast>
  //   ).toBeA11yWhenRendered();
  // });
});
