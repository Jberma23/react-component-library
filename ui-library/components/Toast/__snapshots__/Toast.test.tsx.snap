// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast has a clickable dismiss button 1`] = `<DocumentFragment />`;

exports[`Toast renders with required props 1`] = `
<DocumentFragment>
  <div
    class="w-[fit-content] grid grid-cols-[min-content,1fr,min-content] shadow-1 bg-white z-100 fixed bottom-m right-m transform transition duration-[200ms] opacity-0 translate-y-[100%]"
  >
    <div
      class="p-s"
    >
      <svg
        aria-hidden="true"
        class="fill-current inline fill-success"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Success
        </title>
        <path
          clip-rule="evenodd"
          d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm13.36-1.814a.75.75 0 10-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 00-1.06 1.06l2.25 2.25a.75.75 0 001.14-.094l3.75-5.25z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="py-s"
    >
      Hello World
    </div>
    <button
      class="p-s"
      title="dismiss notification"
    >
      <svg
        aria-hidden="true"
        class="fill-current inline"
        fill="currentColor"
        height="24"
        viewBox="0 0 20 20"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Dismiss
        </title>
        <path
          d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
